### PR TITLE
dockerfile: switch location and release notes

### PR DESCRIPTION
**Pull request by crazy-max**
needs https://github.com/docker/docker.github.io/pull/15151 (for proper redirects)

adds a release notes page for the dockerfile reference. needs to create a subsection for release notes and therefore also needs to change the reference location from `/engine/reference/builder` to `/reference/dockerfile/` as discussed in https://github.com/docker/docker.github.io/pull/14948#issuecomment-1167230746.

preview: https://deploy-preview-15110--docsdocker.netlify.app/reference/dockerfile/

image

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -174,10 +174,10 @@ fetch-remote:
         src:
           - "docs/configuration.md"
 
-  - repo: "https://github.com/moby/buildkit"
+  - repo: "https://github.com/crazy-max/buildkit"
     default_branch: "master"
-    ref: "master"
+    ref: "docs-dockerfile-changelog"
```
Comment (crazy-max) [inline _config.yml:179]: use my fork in the meantime

File: reference/dockerfile/index.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+title: Dockerfile reference
+description: "Dockerfiles use a simple DSL which allows you to automate the steps you would normally manually take to create an image."
+keywords: build, dockerfile, reference
+toc_max: 3
+redirect_from:
+- /engine/reference/builder/
+- /reference/builder/
+---
+
+{% include_relative reference.md %}
```
Comment (crazy-max) [inline reference/dockerfile/index.md:11]: @thaJeztah as discussed I have created a stub file and moved the frontmatter section from buildkit repo to the stub file so we have control over it in docs repo.

**Comment by crazy-max** [comment]
htmlproofer not happy because we need to also update reference location on foreign repos (docker/cli, docker/buildx and distribution/distribution)

```
#16 [htmlproofer 1/1] RUN --mount=type=bind,from=generate,source=/out,target=_site   htmlproofer ./_site     --disable-external     --internal-domains="docs.docker.com,docs-stage.docker.com,localhost:4000"     --file-ignore="/^./_site/engine/api/.*$/,./_site/registry/configuration/index.html"     --url-ignore="/^/docker-hub/api/latest/.*$/,/^/engine/api/v.+/#.*$/,/^/glossary/.*$/"
#16 0.563 Running ["LinkCheck", "ImageCheck", "ScriptCheck"] on ["./_site"] on *.html... 
#16 0.563 
#16 0.563 
#16 173.0 Ran on 1717 files!
#16 173.0 
#16 173.0 
#16 173.0 - ./_site/engine/extend/config/index.html
#16 173.0   *  linking to internal hash #entrypoint that does not exist (line 205)
#16 173.0      <a href="../reference/builder.md#entrypoint"></a>
#16 173.0   *  linking to internal hash #workdir that does not exist (line 210)
#16 173.0      <a href="../reference/builder.md#workdir"></a>
#16 173.0 - ./_site/engine/reference/commandline/build/index.html
#16 173.0   *  linking to internal hash #add that does not exist (line 759)
#16 173.0      <a href="/engine/reference/builder/#add"><em>ADD</em></a>
#16 173.0   *  linking to internal hash #buildkit that does not exist (line 1151)
#16 173.0      <a href="/engine/reference/builder/#buildkit">enable BuildKit</a>
#16 173.0   *  linking to internal hash #buildkit that does not exist (line 844)
#16 173.0      <a href="/engine/reference/builder/#buildkit">BuildKit backend</a>
#16 173.0   *  linking to internal hash #buildkit that does not exist (line 1109)
#16 173.0      <a href="/engine/reference/builder/#buildkit">enable BuildKit</a>
#16 173.0   *  linking to internal hash #copy that does not exist (line 136)
#16 173.0      <a href="/engine/reference/builder/#copy"><em>COPY</em></a>
#16 173.0   *  linking to internal hash #copy that does not exist (line 1071)
#16 173.0      <a href="/engine/reference/builder/#copy"></a>
#16 173.0   *  linking to internal hash #dockerignore-file that does not exist (line 276)
#16 173.0      <a href="/engine/reference/builder/#dockerignore-file">.dockerignore file</a>
#16 173.0   *  linking to internal hash #dockerignore-file that does not exist (line 842)
#16 173.0      <a href="/engine/reference/builder/#dockerignore-file">creating a .dockerignore file</a>
#16 173.0 - ./_site/engine/reference/commandline/buildx_build/index.html
#16 173.0   *  linking to internal hash #automatic-platform-args-in-the-global-scope that does not exist (line 963)
#16 173.0      <a href="/engine/reference/builder/#automatic-platform-args-in-the-global-scope">
#16 173.0 documentation</a>
#16 173.0 - ./_site/engine/reference/commandline/cli/index.html
#16 173.0   *  linking to internal hash #buildkit that does not exist (line 219)
#16 173.0      <a href="../builder.md#buildkit">BuildKit backend</a>
#16 173.0 - ./_site/engine/reference/run/index.html
#16 173.0   *  internally linking to builder.md, which does not exist (line 1808)
#16 173.0      <a href="builder.md"><em>Dockerfile</em></a>
#16 173.0 - ./_site/language/nodejs/build-images/index.html
#16 173.0   *  linking to internal hash #parser-directives that does not exist (line 258)
#16 173.0      <a href="/reference/dockerfile/index.md#parser-directives">Parser directives</a>
#16 173.0   *  linking to internal hash #syntax that does not exist (line 255)
#16 173.0      <a href="/reference/dockerfile/index.md#syntax"> parser directive</a>
#16 173.0 - ./_site/registry/spec/deprecated-schema-v1/index.html
#16 173.0   *  linking to internal hash #from that does not exist (line 141)
#16 173.0      <a href="https://docs.docker.com/engine/reference/builder/#from">Dockerfile reference</a>
#16 173.0 
#16 173.0 HTML-Proofer found 16 failures!
#16 ERROR: process "/bin/sh -c htmlproofer ./_site     --disable-external     --internal-domains=\"docs.docker.com,docs-stage.docker.com,localhost:4000\"     --file-ignore=\"/^./_site/engine/api/.*$/,./_site/registry/configuration/index.html\"     --url-ignore=\"/^/docker-hub/api/latest/.*$/,/^/engine/api/v.+/#.*$/,/^/glossary/.*$/\"" did not complete successfully: exit code: 1
```

File: _config.yml [hunk 3]
```diff
@@ -178,6 +178,6 @@ fetch-remote:
     default_branch: "master"
     ref: "master"
     paths:
-      - dest: "engine/reference/builder.md"
+      - dest: "reference/dockerfile"
```
Comment (thaJeztah) [inline _config.yml:181]: Did these need the trailing slash? (to prevent extra redirects?)

```suggestion
      - dest: "reference/dockerfile/"
```

Comment (crazy-max) [reply _config.yml:181]: Yes indeed, I just reopened this PR to remember it exists tbh :sweat_smile: but would love to have it merged. I think we still need to figure out how and when we want to move `/engine/reference` to `/reference` first.

**Pull request closed without merging**